test/controllers/desks_controller_test.rb

```diff
@@ -44,14 +46,13 @@ class DesksControllerTest < ActionDispatch::IntegrationTest
   end
 
   test "#update updates desk and responds with updated desk" do
-    put desk_path(@desk), params: { desk: { occupied: false } }
+    put desk_path(@desk, format: :json), params: { desk: { occupied: false } }
 
-    assert_response :success
-    assert_equal false, JSON.parse(response.body)["occupied"]
+    assert_response 204
```

Comment:
Assert that `occupied` is the only thing that changed.

Maybe something like `assert_equal @desk.merge(occupied: false), JSON.parse(response.body)`